### PR TITLE
add PEC dataset

### PR DESCRIPTION
**Pull request by zhongpeixiang**
A persona-based empathetic conversation dataset published at EMNLP 2020.

### CODE REVIEW

File: datasets/pec/pec.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                                "context": context,
+                                "response_speaker": response_speaker,
+                                "response": response,
+                            }
+                            context_speakers = []
+                            context = []
+                            response_speaker = ""
+                            response = ""
+                            example_id += 1
+                    except (IndexError, KeyError):
```
Comment (lhoestq) [inline datasets/pec/pec.py:202]: Could you remove that if it was used for debugging purposes ?

File: datasets/pec/pec.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        return persona
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: Downloads the data and defines the splits
+        # dl_manager is a datasets.download.DownloadManager that can be used to
+        # download and extract URLs
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_dir = os.path.join(dl_dir, "hf_pec")
+        print("The downloaded data has been saved to ", data_dir)
```
Comment (lhoestq) [inline datasets/pec/pec.py:114]:
```suggestion

```

File: datasets/pec/pec.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        else:
+            filepath = [filepath]
+
+        # create persona
+        if not hasattr(self, "persona_" + self.config.domain):
+            persona_paths = [os.path.join(os.path.dirname(fpath), "persona.txt") for fpath in filepath]
+            setattr(self, "persona_" + self.config.domain, self._load_persona(persona_paths))
+
+        persona = getattr(self, "persona_" + self.config.domain)
+        # print("Domain {0}: number of unique personas: {1}".format(self.config.domain, len(persona)))
```
Comment (lhoestq) [inline datasets/pec/pec.py:171]: Why not simply use this instead : 
```
persona_paths = [os.path.join(os.path.dirname(fpath), "persona.txt") for fpath in filepath]
persona = self._load_persona(persona_paths)
```
?

(or maybe I'm missing something)

**Review by lhoestq — COMMENTED** [review]
Nice job ! This is super clean thanks

I left a few minor comments and I think after they're resolved we'll be ready to merge :)

File: datasets/pec/pec.py (hunk 3 above)
Comment (zhongpeixiang) [reply datasets/pec/pec.py:171]: I dont want to re-create the `persona` for every split of the dataset since it is shared by all splits.

**Comment by zhongpeixiang** [comment]
The checks failed again even if I didn't make any changes.

**Comment by lhoestq** [comment]
you just need to rebase from master to fix the CI :)

File: datasets/pec/pec.py (hunk 3 above)
Comment (lhoestq) [reply datasets/pec/pec.py]: Then feel free to generate the persona once in the `_split_generator` method (it's called once for all splits), and pass the persona object through the `gen_kwargs` that are passed to `_generate_examples `
This way the persona will be created only once and reused in all the splits.
Does that make sense ?

**Comment by zhongpeixiang** [comment]
Sorry for the mess, I'm confused by the rebase and thus created a new branch.

**Pull request closed without merging**